sktime/classification/distance_based/_time_series_neighbors_tslearn.py

```diff
@@ -0,0 +1,125 @@
+          will have a greater influence than neighbors which are further away.
+        - [callable] : a user-defined function which accepts an array of
+          distances, and returns an array of the same
+          shape containing the weights.
+
+    metric : {'dtw', 'softdtw', 'ctw', 'euclidean', 'sqeuclidean', 'cityblock',  'sax'}
+        default: 'dtw'.
+        Metric to be used at the core of the nearest neighbor procedure.
+        When ``'sax'`` is provided as a metric, the data is expected to be
+        normalized such that each time series has zero mean and unit
+        variance. ``'euclidean'``, ``'sqeuclidean'``, ``'cityblock'`` are described
+        in `scipy.spatial.distance doc
+        <https://docs.scipy.org/doc/scipy/reference/spatial.distance.html>`_.
+
+    metric_params : dict or None (default: None)
+        Dictionary of metric parameters.
+        For metrics that accept parallelization of the cross-distance matrix
+        computations, ``n_jobs`` and ``verbose`` keys passed in ``metric_params``
+        are overridden by the ``n_jobs`` and ``verbose`` arguments.
+        For ``'sax'`` metric, these are hyper-parameters to be passed at the
+        creation of the ``SymbolicAggregateApproximation`` object.
+
+    n_jobs : int or None, optional (default=None)
+        The number of jobs to run in parallel for cross-distance matrix
+        computations.
+        Ignored if the cross-distance matrix cannot be computed using
+        parallelization.
+        ``None`` means 1 unless in a ``joblib.parallel_backend`` context.
+        ``-1`` means using all processors.
+
+    verbose : int, optional (default=0)
+        The verbosity level: if non zero, progress messages are printed.
+        Above 50, the output is sent to stdout.
+        The frequency of the messages increases with the verbosity level.
+        If it more than 10, all iterations are reported.
+    """
+
+    _tags = {
+        # packaging info
+        # --------------
+        "authors": "fkiraly",
+        "python_dependencies": "tslearn",
+        # estimator type
+        # --------------
+        "capability:multivariate": True,
+        "capability:unequal_length": True,
+    }
+
+    # defines the name of the attribute containing the tslearn estimator
+    _estimator_attr = "_tslearn_knn"
+
+    def _get_tslearn_class(self):
+        """Get tslearn class.
+
+        should import and return tslearn class
+        """
+        from tslearn.neighbors import KNeighborsTimeSeriesClassifier as TsLearnKnn
+
+        return TsLearnKnn
+
+    def __init__(
+        self,
+        n_neighbors=5,
+        weights="uniform",
+        metric="dtw",
+        metric_params=None,
+        n_jobs=None,
+        verbose=0,
+    ):
+        self.n_neighbors = n_neighbors
+        self.weights = weights
+        self.metric = metric
+        self.metric_params = metric_params
+        self.n_jobs = n_jobs
+        self.verbose = verbose
```

Comment:
Comment based on no previous familiarity, so sorry if it's meaningless: do you not need to call super init?

Reply:
yes, thanks - that's also what the tests are telling us.